### PR TITLE
Wait for online text in test dependencies

### PR DESCRIPTION
**Pull request by hugoesthere**

### CODE REVIEW

File: exosphere-shared/src/docker-helper.ls [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
       | err                              => done err
       | running-containers.includes name => done!
       | otherwise                        =>
         docker.create-container {Image, name, HostConfig}, (err, container) -> 
           | err => done err
-          container.start (err, data) ->
-            | err => done err
-            wait 2_000, done
+          container.attach {stream: true, stdout: true, stederr: true}, (err, stream) ->
+            stream = new TextStreamSearch stream
```
Comment (charlierudolph) [inline exosphere-shared/src/docker-helper.ls]: can this move down to line 23 so its created just before its used? Could also join the two lines into one
```livescript
new TextStreamSearch(stream).wait online-text, done
```

File: exosphere-shared/src/docker-helper.ls [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
       | running-containers.includes name => done!
       | otherwise                        =>
         docker.create-container {Image, name, HostConfig}, (err, container) -> 
           | err => done err
-          container.start (err, data) ->
-            | err => done err
-            wait 2_000, done
+          container.attach {stream: true, stdout: true, stederr: true}, (err, stream) ->
+            stream = new TextStreamSearch stream    
+            container.start (err, data) ->
```
Comment (charlierudolph) [inline exosphere-shared/src/docker-helper.ls]: data appears unused

**Review by charlierudolph — APPROVED** [review]

File: exosphere-shared/src/docker-helper.ls (hunk 1 above)
Comment (hugoesthere) [reply exosphere-shared/src/docker-helper.ls]: Is it possible then to miss data that's been passed through the stream between calling `container.start` and when `TextStreamSearch` is constructed in the callback of `container.start`? I figured attaching to the stream before starting the container would ensure capturing all output from the process.

Comment (charlierudolph) [reply exosphere-shared/src/docker-helper.ls]: Auhh correct. Can we avoid overriding the `stream` variable and name this `testStreamSearch`?

Comment (hugoesthere) [reply exosphere-shared/src/docker-helper.ls]: Yes, I like that

**Pull request closed without merging**